proxy_map.go

```diff
@@ -997,6 +1024,24 @@ func (m *Map) tryRemove(ctx context.Context, key interface{}, timeout int64) (in
 	}
 }
 
+func (m *Map) aggregate(ctx context.Context, req *proto.ClientMessage, decoder func(message *proto.ClientMessage) *serialization.Data) (interface{}, error) {
+	resp, err := m.invokeOnRandomTarget(ctx, req, nil)
+	if err != nil {
+		return nil, err
+	}
+	data := decoder(resp)
+	obj, err := m.convertToObject(data)
+	if err != nil {
+		return nil, err
+	}
+	// if this is a canonicalizing set return a slice of interface[}
```

Comment:
We might need to update the comment now

Reply:
OK: https://github.com/hazelcast/hazelcast-go-client/pull/589/commits/5d27f239bdc4f73739563952479e1dfbc00ccbfe